packages/gtmcore/gtmcore/environment/repository.py

```diff
@@ -1,9 +1,87 @@
 import os
 import pickle
-from typing import (Any, List, Dict)
+import functools
+from typing import (Any, Dict, List, Optional)
 from operator import itemgetter
 
+import redis_lock
+from redis import StrictRedis
+
+from gtmcore.logging import LMLogger
 from gtmcore.configuration import Configuration
+from gtmcore.exceptions import GigantumLockedException
+
+class RepositoryLock(object):
+    def __init__(self, config_file: Optional[str] = None, key: Optional[str] = None):
+        self.client_config = Configuration(config_file)
+
+        self.key = key or "repository_updates"
```

Comment:
```suggestion
        self.key = key or "base_repository_updates"
```
Might be a bit more clear and differentiate from normal lock behavior that is protecting project and dataset repositories?